### PR TITLE
CQL types

### PR DESCRIPTION
**Pull request by asledz**
Add missing CQL types. 

Add missing CQL types: blob, date, time, float and doubles et.c.

(some types are yet to be commited - don't merge yet)

### CODE REVIEW

**Comment by asledz** [comment]
I will upload full test later today.
Types are ready to review :) @piodul @psarna

File: scylla/src/frame/value.rs [hunk 1]
```diff
@@ -205,6 +244,19 @@ impl Value for String {
...
 }
 
+impl Value for Vec<u8> {
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+        let val_len: i32 = self.len().try_into().map_err(|_| ValueTooBig)?;
+        buf.put_i32(val_len);
+
+        for byte in self {
+            buf.put_u8(*byte);
+        }
```
Comment (piodul) [inline scylla/src/frame/value.rs:254]: Use `extend_from_slice` here - it is more idiomatic and might have better performance:

```rust
buf.extend_from_slice(&self);
```

File: scylla/src/frame/cql_types_test.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+#[ignore]
+async fn test_cql_collections() {
+    // Create connection
+    let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+
+    println!("Connecting to {} ...", uri);
+
+    let session = crate::transport::session::Session::connect(uri, None)
+        .await
+        .unwrap();
```
Comment (piodul) [inline scylla/src/frame/cql_types_test.rs:12]: We are now using `SessionBuilder` to construct sessions. The example won't compile after merging this branch to `main` branch. Please rebase onto current `main` and fix.

File: scylla/src/frame/response/cql_to_rust.rs [hunk 3]
```diff
@@ -68,6 +68,10 @@ macro_rules! impl_from_cql_val {
 }
 
 impl_from_cql_val!(i32, as_int); // i32::from_cql<CQLValue>
+impl_from_cql_val!(u32, as_date); // u32::from_cql<CQLValue>
+impl_from_cql_val!(f32, as_float); // i32::from_cql<CQLValue>
+impl_from_cql_val!(f64, as_double); // u32::from_cql<CQLValue>
+impl_from_cql_val!(bool, as_boolean); // u32::from_cql<CQLValue>
```
Comment (piodul) [inline scylla/src/frame/response/cql_to_rust.rs:74]: Please change `u32`/`i32` in the comments to correct types.

Comment (asledz) [reply scylla/src/frame/response/cql_to_rust.rs]: Fixed, I must have left it by mistake.

File: scylla/src/frame/cql_types_test.rs (hunk 2 above)
Comment (asledz) [reply scylla/src/frame/cql_types_test.rs]: Fixed.

File: scylla/src/frame/value.rs (hunk 1 above)
Comment (asledz) [reply scylla/src/frame/value.rs]: Didn't think about it, makes sense! Fixed it.

File: scylla/src/frame/value.rs [hunk 4]
```diff
@@ -187,6 +187,45 @@ impl Value for i64 {
...
+        buf.put_u32(*self);
+        Ok(())
+    }
+}
+
+impl Value for bool {
+    fn serialize(&self, buf: &mut Vec<u8>) -> Result<(), ValueTooBig> {
+        buf.put_i32(1);
+        let false_bytes: &[u8] = &[0x00];
+        let true_bytes: &[u8] = &[0x00];
```
[the remark this answers is not in the document]
Comment (asledz) [reply scylla/src/frame/value.rs]: My mistake, fixed.

File: scylla/src/frame/response/result.rs [hunk 5]
```diff
@@ -347,6 +407,51 @@ fn deser_cql_value(typ: &ColumnType, buf: &mut &[u8]) -> StdResult<CQLValue, Par
...
+                    "Buffer length should be 4 not {}",
+                    buf.len()
+                )));
+            }
+            CQLValue::Date(buf.read_u32::<BigEndian>()?)
+        }
+        Double => {
+            if !buf.len() != 8 {
+                return Err(ParseError::BadData(format!(
+                    "Buffer length should be 4 not {}",
```
[the remark this answers is not in the document]
Comment (asledz) [reply scylla/src/frame/response/result.rs]: Fixed.

File: scylla/src/frame/response/result.rs [hunk 6]
```diff
@@ -347,6 +407,51 @@ fn deser_cql_value(typ: &ColumnType, buf: &mut &[u8]) -> StdResult<CQLValue, Par
...
+            if !buf.len() == 0 {
+                return Err(ParseError::BadData(format!(
+                    "Buffer length should not be {}",
+                    buf.len()
+                )));
+            }
+            CQLValue::Blob(buf.to_vec())
+        }
+        Date => {
+            if !buf.len() != 4 {
```
[the remark this answers is not in the document]
Comment (asledz) [reply scylla/src/frame/response/result.rs]: Done, thanks!

File: scylla/src/frame/response/result.rs [hunk 7]
```diff
@@ -347,6 +407,51 @@ fn deser_cql_value(typ: &ColumnType, buf: &mut &[u8]) -> StdResult<CQLValue, Par
...
+            if !buf.len() != 1 {
+                return Err(ParseError::BadData(format!(
+                    "Buffer length should be 1 not {}",
+                    buf.len()
+                )));
+            }
+            CQLValue::Boolean(buf[0] != 0x00)
+        }
+        Blob => {
+            if !buf.len() == 0 {
```
[the remark this answers is not in the document]
Comment (asledz) [reply scylla/src/frame/response/result.rs]: Done.

File: scylla/src/frame/response/result.rs [hunk 8]
```diff
@@ -593,7 +585,7 @@ fn deser_rows(buf: &mut &[u8]) -> StdResult<Rows, ParseError> {
     })
 }
 
-#[allow(clippy::unnecessary_wraps)]
+#[allow(clippy::unnecessary_unwrap)]
```
Comment (piodul) [inline scylla/src/frame/response/result.rs:588]: Clippy does not complain here about [`unnecessary_unwrap`](https://rust-lang.github.io/rust-clippy/master/#unnecessary_unwrap), but [`unnecesary_wraps`](https://rust-lang.github.io/rust-clippy/master/#unnecessary_wraps). A similar name, but a different issue. Please change the allowed lint name back.

Please also do it for `deser_schema_change`.

**Review by piodul — CHANGES_REQUESTED** [review]
Unfortunately, the test doesn't work for me:

```
[piodul@localhost scylla-rust-driver]$ cargo test test_cql_types -- --ignored
   Compiling scylla v0.0.1 (/home/piodul/code/scylla-rust-driver/scylla)
    Finished test [unoptimized + debuginfo] target(s) in 1.31s
     Running target/debug/deps/scylla-ec97baa3bb679560

running 1 test
test frame::cql_types_test::test_cql_types ... FAILED

failures:

---- frame::cql_types_test::test_cql_types stdout ----
Connecting to 127.0.0.1:9042 ...
thread 'frame::cql_types_test::test_cql_types' panicked at 'called `Result::unwrap()` on an `Err` value: ProtocolError("Error parsing message")', scylla/src/frame/cql_types_test.rs:34:77
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    frame::cql_types_test::test_cql_types

test result: FAILED. 0 passed; 1 failed; 0 ignored; 0 measured; 65 filtered out; finished in 0.03s

error: test failed, to rerun pass '-p scylla --lib'
```

File: scylla/src/frame/response/result.rs (hunk 8 above)
Comment (asledz) [reply scylla/src/frame/response/result.rs:588]: so sorry!
locally, clippy complained that ` unknown clippy lint: clippy::unnecessary_wraps`, so I just automatically corrected it - gonna change the lint name back rn

**Comment by asledz** [comment]
Moved everything to new PR ( #175 )

**Pull request closed without merging**